superduperdb/components/model.py

```diff
@@ -538,6 +538,7 @@ def predict(self, dataset: t.Union[t.List, QueryDataset]) -> t.List:
     def _prepare_select_for_predict(self, select, db):
         if isinstance(select, dict):
             select = Document.decode(select).unpack()
+        select.set_db(db)
```

Comment:
why?

Reply:
https://github.com/SuperDuperDB/superduperdb/pull/2111/files#r1618342992

Keep the same logic with this